pkg/api/validation/validation_test.go

```diff
@@ -2881,6 +2928,116 @@ func TestAlphaHugePagesIsolation(t *testing.T) {
 	}
 }
 
+func TestAlphaPVCVolumeMode(t *testing.T) {
+	// Enable alpha feature BlockVolumeSupport for PVC
+	err := utilfeature.DefaultFeatureGate.Set("BlockVolumeSupport=true")
+	if err != nil {
+		t.Errorf("Failed to enable feature gate for BlockVolumeSupport: %v", err)
+		return
+	}
+
+	// Success Cases
+	successCasesPVC := map[string]*api.PersistentVolumeClaim {
+		"valid block value":      CreateTestVolModePVC(api.PersistentVolumeBlock),
+		"valid filesystem value": CreateTestVolModePVC(api.PersistentVolumeFilesystem),
+	}
+	for k, v := range successCasesPVC {
+		if errs := ValidatePersistentVolumeClaim(v); len(errs) != 0 {
+			t.Errorf("expected success for %s", k)
+		}
+	}
+
+	// Error Cases
+	errorCasesPVC := map[string]*api.PersistentVolumeClaim {
+		"invalid value": CreateTestVolModePVC(api.PersistentVolumeMode("fake")),
+		"empty value":   CreateTestVolModePVC(api.PersistentVolumeMode("")),
+	}
+	for k, v := range errorCasesPVC {
+		if errs := ValidatePersistentVolumeClaim(v); len(errs) == 0 {
+			t.Errorf("expected failure for %s", k)
+		}
+	}
+}
+
+func TestAlphaPVVolumeMode(t *testing.T) {
+	// Enable alpha feature BlockVolumeSupport for PV
+	err := utilfeature.DefaultFeatureGate.Set("BlockVolumeSupport=true")
+	if err != nil {
+		t.Errorf("Failed to enable feature gate for BlockVolumeSupport: %v", err)
+		return
+	}
+
+	// Success Cases
+	successCasesPV := map[string]*api.PersistentVolume {
+		"valid block value":      CreateTestVolModePV(api.PersistentVolumeBlock),
+		"valid filesystem value": CreateTestVolModePV(api.PersistentVolumeFilesystem),
```

Comment:
nil volume mode is also a success case.

Reply:
what do you mean by nil volume mode?  where volumeMode is not included at all in the Spec? If so that is already tested in each of the success cases in the pv and pvc tests no?

Reply:
Yes I mean where it's not specified + the feature gate is enabled.

Reply:
or do you want me to explicitly pass in a spec with no volumeMode just to keep it in test block?

Reply:
Yeah maybe the CreateTestVolModePV can take in a ptr argument instead.